tests/cupyx_tests/test_lapack.py

```diff
@@ -30,36 +32,59 @@ def _make_matrix(self, shape, alpha, beta):
 
     def setUp(self):
         self.dtype = numpy.dtype(self.dtype)
-        if self.dtype.char in 'fF':
-            self.r_dtype = numpy.float32
-        else:
-            self.r_dtype = numpy.float64
         n = self.n
         nrhs = 1 if self.nrhs is None else self.nrhs
         # Diagonally dominant matrix is used as it is stable
         alpha = 2.0 / n
         a = self._make_matrix((n, n), alpha, -alpha / 2)
-        diag = cupy.diag(cupy.ones((n,), dtype=self.r_dtype))
+        diag = cupy.diag(cupy.ones((n,), dtype=self.dtype.char.lower()))
         a[diag > 0] = 0
         a += diag
         x = self._make_matrix((n, nrhs), 0.2, 0.9)
         b = cupy.matmul(a, x)
         b_shape = [n]
         if self.nrhs is not None:
             b_shape.append(nrhs)
+        b = b.reshape(b_shape)
         self.a = a
-        self.b = b.reshape(b_shape)
+        if self.nrhs is None or self.nrhs == 1:
+            self.b = b.copy(order=self.order)
+        else:
+            self.b = b.copy(order='F')
         self.x_ref = x.reshape(b_shape)
-        if self.r_dtype == numpy.float32:
-            self.tol = self._tol['f']
-        elif self.r_dtype == numpy.float64:
-            self.tol = self._tol['d']
+        self.tol = self._tol[self.dtype.char.lower()]
 
     def test_gesv(self):
-        x = lapack.gesv(self.a, self.b)
-        cupy.testing.assert_allclose(x, self.x_ref,
+        lapack.gesv(self.a, self.b)
+        cupy.testing.assert_allclose(self.b, self.x_ref,
                                      rtol=self.tol, atol=self.tol)
 
+    def test_invalid_cases(self):
+        if self.nrhs is None or self.nrhs == 1:
+            raise unittest.SkipTest()
+        with pytest.raises(ValueError):
+            ng_a = self.a.reshape(1, self.n, self.n)
```

Comment:
Could you create `ng_a` (and `ng_b`) outside `with pytest.raises`?

Reply:
OK, I will move them outside `with pytest.raise`.